src/plot/highlight.js

```diff
@@ -0,0 +1,300 @@
+    const {onBrush, onDrag, drag} = this.props;
+    const {brushing, dragging} = this.state;
+    const {xLoc, yLoc} = getLocs(e.nativeEvent);
+    if (brushing) {
+      const brushArea = this._getDrawArea(xLoc, yLoc);
+      this.setState({brushArea});
+
+      if (onBrush) {
+        onBrush(this._convertAreaToCoordinates(brushArea));
+      }
+    }
+
+    if (drag && dragging) {
+      const brushArea = this._getDragArea(xLoc, yLoc);
+      this.setState({brushArea});
+      if (onDrag) {
+        onDrag(this._convertAreaToCoordinates(brushArea));
+      }
+    }
+  }
+
+  render() {
+    const {
+      color,
+      className,
+      highlightHeight,
+      highlightWidth,
+      highlightX,
+      highlightY,
+      innerWidth,
+      innerHeight,
+      marginLeft,
+      marginRight,
+      marginTop,
+      marginBottom,
+      opacity
+    } = this.props;
+    const {brushArea: {left, right, top, bottom}} = this.state;
+
+    let leftPos = 0;
+    if (highlightX) {
+      const xScale = getAttributeScale(this.props, 'x');
+      leftPos = xScale(highlightX);
+    }
+
+    let topPos = 0;
+    if (highlightY) {
+      const yScale = getAttributeScale(this.props, 'y');
+      topPos = yScale(highlightY);
+    }
+
+    const plotWidth = (marginLeft + marginRight) + innerWidth;
+    const plotHeight = (marginTop + marginBottom) + innerHeight;
+    const touchWidth = highlightWidth || plotWidth;
+    const touchHeight = highlightHeight || plotHeight;
+
+    return (
+      <g
+        transform={`translate(${leftPos}, ${topPos})`}
+        className={`${className} rv-highlight-container`}
+        >
+        <rect
+          className="rv-mouse-target"
+          fill="black"
+          opacity="0"
+          x="0"
+          y="0"
+          width={Math.max(touchWidth, 0)}
+          height={Math.max(touchHeight, 0)}
+          onMouseDown={e => this.startBrushing(e)}
+          onMouseMove={e => this.onBrush(e)}
+          onMouseUp={e => this.stopBrushing(e)}
+          onMouseLeave={e => this.stopBrushing(e)}
+          // preventDefault() so that mouse event emulation does not happen
+          onTouchEnd={e => {
+            e.preventDefault();
+            this.stopBrushing(e);
+          }}
+          onTouchCancel={e => {
+            e.preventDefault();
+            this.stopBrushing(e);
+          }}
+          onContextMenu={e => e.preventDefault()}
+          onContextMenuCapture={e => e.preventDefault()}
+        />
+        <rect
+          className="rv-highlight"
+          pointerEvents="none"
+          opacity={opacity}
+          fill={color}
+          x={left}
+          y={top}
+          width={Math.min(Math.max(0, right - left), touchWidth)}
+          height={Math.min(Math.max(0, bottom - top), touchHeight)}
+        />
+      </g>
+    );
+  }
+}
+
+Highlight.displayName = 'HighlightOverlay';
+Highlight.defaultProps = {
+  color: 'rgb(77, 182, 172)',
+  className: '',
+  enableX: false,
+  enableY: false,
```

Comment:
minor nit - this is probably going to create a bunch of GitHub issues where people are like "I added Highlight but nothing happens??" because by default, the component doesn't do anything. My .02 would be to default these values to `true`, so someone can just drop this in and get highlighting in all directions.

Reply:
You are 100% correct, updating